### PR TITLE
Improving architecture selection with hiera

### PR DESCRIPTION
**Pull request by ody**
Implements the use of hiera to build out architecture profiles instead
of attempting to encode it directly into modules using conditional
expressions

### CODE REVIEW

File: hiera.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+version: 5
+
+defaults:
+  datadir: hieradata
```
Comment (mcka1n) [inline hiera.yaml]: I'm just wondering here if the `datadir` should be `hiera_data` ... or is there any convention for the hiera data folder name? 🤔

**Comment by mcka1n** [comment]
Hey @ody 

I left one question above (nothing deal breaker)

But other than that this PR looks good me!

File: hiera.yaml (hunk 1 above)
Comment (ody) [reply hiera.yaml]: Could probably just use what is used in the skeleton control-repo.

https://github.com/puppetlabs/control-repo/blob/production/hiera.yaml

**Comment by ody** [comment]
@mcka1n Updated commit to mirror control-repo. Thanks for review!

**Comment by mcka1n** [comment]
@ody looks good to me 👍

File: hiera.yaml (hunk 1 above)
Comment (mcka1n) [reply hiera.yaml]: Thanks for the reference to the control-repo!

**Review by mcka1n — APPROVED** [review]

**Pull request merged**